src/current/v23.2/known-limitations.md

```diff
@@ -10,100 +10,61 @@ This page describes newly identified limitations in the CockroachDB {{page.relea
+  - PL/pgSQL does not support `NOT NULL` variable declarations. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/105243)
 
-### Common table expressions are not supported in user-defined functions
+#### Limitations in User-Defined Functions (UDFs) and Stored Procedures
 
-[Common table expressions]({% link {{ page.version.version }}/common-table-expressions.md %}) (CTE), recursive or non-recursive, are not supported in [user-defined functions]({% link {{ page.version.version }}/user-defined-functions.md %}) (UDF). That is, you cannot use a `WITH` clause in the body of a UDF.
+  - PL/pgSQL does not support the `PERFORM` keyword. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/108416)
+  - UDFs and stored procedures cannot call other UDFs or stored procedures from within their bodies. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/88198)
+  - The `setval` function cannot be resolved when used inside UDF bodies. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/110860)
+  - DDL statements (e.g., `CREATE TABLE`, `CREATE INDEX`) are not allowed within UDFs or procedures. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/110080)
+  - UDF and stored procedure definitions do not support `OUT` and `INOUT` argument modes. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/100405)
 
-[Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/92961)
+### SQL Optimizer and Read Committed Isolation
 
-### Low estimated Request Units are rounded to zero
+#### Optimizer and Locking Behavior
 
-The [Request Units](https://www.cockroachlabs.com/docs/cockroachcloud/plan-your-cluster-serverless#request-units) (RUs) estimate surfaced in [`EXPLAIN ANALYZE`]({% link {{ page.version.version }}/explain-analyze.md %}) is displayed as an integer value. Because of this, fractional RU estimates, which represent very inexpensive queries, are rounded down to zero.
+- The SQL optimizer has limitations under certain isolation levels:
+  - The new implementation of `SELECT FOR UPDATE` is not yet the default setting under `SERIALIZABLE` isolation. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/114737)
+  - `SELECT FOR UPDATE` does not lock completely-`NULL` column families in multi-column-family tables. [Tracking GitHub issue](https://github.com/cockroachdb/cockroach/issues/116836)
+  - `SELECT ... FOR UPDATE` and `SELECT ... FOR SHARE` locks are dropped on lease transfers and range splits/merges. Unreplicated locks, used by default under SERIALIZABLE isolation, can be dropped during such events, affecting the desired ordering of concurrent accesses and potentially leading to transaction retry errors. [More information on `SELECT ... FOR UPDATE`](https://www.cockroachlabs.com/docs/v23.2/select-for-update)
```

Comment:
This isn't a new limitation, it has been true since at least 21.2, maybe earlier (whenever locking was initially added). What is new in 23.2 is that this is no longer a limitation either (a) when using read committed isolation or (b) with `enable_durable_locking_for_serializable` set to true when using serializable isolation.

Reply:
I'm going to remove this item, because I had already updated the existing limitation in #18098 to reflect this info. cc @mikeCRL